In [1]:
import numpy as np
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
df.tail()

station        date  prcp  tobs
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

In [4]:
df.date.dtype

dtype('O')

In [5]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [6]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])

In [7]:
# Drop the date column
df = df.drop(columns='date')

In [8]:
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [9]:
from scipy import stats

In [10]:
# Filter data for desired months
jun_data = df[df.index.month == 6]
dec_data = df[df.index.month == 12]

In [11]:
jun_data.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [12]:
dec_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [13]:
# Create collections of temperature data
jun_temp = jun_data.tobs
dec_temp = dec_data.tobs

In [14]:
# Run paired t-test 
stats.ttest_ind(jun_temp, dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

Across all the stations, the mean temperatures in June and December temperature in years 2010-2017 differ by 3.9 degrees Celsius. An unpaired t-test was conducted, and with an extremely low p-value, the difference is deemed statistically significant.